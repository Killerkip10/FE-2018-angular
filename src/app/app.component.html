<app-header></app-header>

<mat-drawer-container class="container">
  <mat-drawer *ngIf="tokenService.getToken()" mode="side" opened>
    <mat-nav-list>
      <a mat-list-item [routerLink]="['user-profile']" routerLinkActive="active">Profile</a>
      <a mat-list-item [routerLink]="['user-list']" routerLinkActive="active">Users list</a>

    </mat-nav-list>
  </mat-drawer>

  <mat-drawer-content>
    <router-outlet></router-outlet>
  </mat-drawer-content>
</mat-drawer-container>

<app-footer></app-footer>
