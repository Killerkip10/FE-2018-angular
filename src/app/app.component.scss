app-header{
  display: block;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 2;
}
.container {
  margin-top: 56px;
  min-height: calc(100vh - 112px);
  mat-drawer{
    position: fixed;
    top: 56px;
    bottom: inherit;
  }
  .active {
    background-color: rgba(0,0,0,.04);
  }
}
