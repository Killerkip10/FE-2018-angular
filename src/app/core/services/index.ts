export * from './auth.service';
export * from './profile.service';
export * from './token.service';
export * from './search-user.service';
