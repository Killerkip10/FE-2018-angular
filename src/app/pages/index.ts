export * from './user-profile/user-profile-page.module';
export * from './login/login-page.module';
export * from './forgot-password/forgot-password-page.module';
export * from './user-list/user-list.module';
