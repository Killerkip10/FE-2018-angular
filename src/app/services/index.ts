export * from './profile.service';
export * from './token.service';
