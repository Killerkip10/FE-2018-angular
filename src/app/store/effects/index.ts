import {ProfileEffects} from './profile';
import {UsersEffects} from './users';

export const effects = [
  ProfileEffects,
  UsersEffects
];
