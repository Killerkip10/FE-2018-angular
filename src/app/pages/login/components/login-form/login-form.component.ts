import {Component} from '@angular/core';
import {Router} from '@angular/router';

import {Store} from '@ngrx/store';
import {GetProfile} from '../../../../store/actions/profile';

import {AuthService} from '../../../../core/services';

@Component({
  selector: 'app-login-form',
  templateUrl: './login-form.component.html',
  styleUrls: ['./login-form.component.scss']
})
export class LoginFormComponent {
  public errorMessage = '';

  constructor(
    private authService: AuthService,
    private router: Router,
    private store: Store<null>
  ) { }

  public submit(loginForm) {
    if (!loginForm.valid) { return; }

    this.authService.login(loginForm.value)
      .subscribe(
        () => {
          this.store.dispatch(new GetProfile());
          this.router.navigate(['/']);
        },
        err => this.errorMessage = err
      );
  }
}
