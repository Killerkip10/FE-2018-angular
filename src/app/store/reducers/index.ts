import {ActionReducerMap} from '@ngrx/store';

import * as fromUsers from './users';
import * as fromProfile from './profile';

export interface State {
  users: fromUsers.State;
  profile: fromProfile.State;
}

export const reducers: ActionReducerMap<State> = {
  users: fromUsers.reducer,
  profile: fromProfile.reducer
};
