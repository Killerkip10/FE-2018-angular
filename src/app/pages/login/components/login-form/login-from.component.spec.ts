import {async, ComponentFixture, TestBed} from '@angular/core/testing';
import {NO_ERRORS_SCHEMA} from '@angular/core';
import {TranslateModule} from '@ngx-translate/core';
import {RouterTestingModule} from '@angular/router/testing';
import {FormsModule} from '@angular/forms';
import {APP_BASE_HREF, CommonModule} from '@angular/common';
import {HttpClientModule} from '@angular/common/http';

import {StoreModule} from '@ngrx/store';
import {EffectsModule} from '@ngrx/effects';
import {reducers} from '../../../../store/reducers';
import {effects} from '../../../../store/effects';

import {LoginFormComponent} from './login-form.component';
import {AppTranslate} from '../../../../app.translate';
import {AuthService} from '../../../../core/services';

describe('LoginFormComponent', () => {
  let component: LoginFormComponent;
  let fixture: ComponentFixture<LoginFormComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [
        HttpClientModule,
        CommonModule,
        RouterTestingModule,
        TranslateModule,
        FormsModule,

        StoreModule.forRoot(reducers),
        EffectsModule.forRoot(effects),

        AppTranslate
      ],
      declarations: [
        LoginFormComponent
      ],
      providers: [
        AuthService,
        {provide: APP_BASE_HREF, useValue: '/'}
      ],
      schemas: [
        NO_ERRORS_SCHEMA
      ]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(LoginFormComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('snapshot', () => {
    component.errorMessage = 'ERROR-TEST';
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
