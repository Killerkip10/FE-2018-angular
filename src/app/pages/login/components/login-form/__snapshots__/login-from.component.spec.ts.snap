// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginFormComponent snapshot 1`] = `
<app-login-form
  authService={[Function AuthService]}
  errorMessage=""
  router={[Function Router]}
  store={[Function Store]}
>
  <section>
    <mat-card>
      <mat-card-title>
        <mat-toolbar
          color="primary"
        >
          SIGN-IN
        </mat-toolbar>
      </mat-card-title>
      <mat-card-content>
        <form
          class="ng-untouched ng-pristine ng-valid"
          novalidate=""
        >
          <div>
            <mat-form-field>
              <input
                class="ng-untouched ng-pristine ng-valid"
                matinput=""
                name="login"
                ng-reflect-model=""
                ng-reflect-name="login"
                ng-reflect-required=""
                ngmodel=""
                placeholder="LOGIN"
                required=""
                type="text"
              />
              <mat-icon
                matsuffix=""
              >
                account_circle
              </mat-icon>
            </mat-form-field>
          </div>
          <div>
            <mat-form-field>
              <input
                class="ng-untouched ng-pristine ng-valid"
                matinput=""
                name="password"
                ng-reflect-model=""
                ng-reflect-name="password"
                ng-reflect-required=""
                ngmodel=""
                placeholder="PASSWORD"
                required=""
                type="password"
              />
              <mat-icon
                matsuffix=""
              >
                lock
              </mat-icon>
            </mat-form-field>
          </div>
          
          <div
            class="actions"
          >
            <button
              color="primary"
              mat-flat-button=""
              type="submit"
            >
               SIGN-IN 
            </button>
            <a
              href="/forgot-password"
              mat-button=""
              ng-reflect-router-link="/forgot-password"
            >
              FORGOT-PASSWORD
            </a>
          </div>
          <mat-divider />
          <button
            class="sign-up"
            mat-button=""
            type="button"
          >
             SIGN-UP 
          </button>
        </form>
      </mat-card-content>
    </mat-card>
  </section>
</app-login-form>
`;

exports[`LoginFormComponent snapshot 2`] = `
<app-login-form
  authService={[Function AuthService]}
  errorMessage={[Function String]}
  router={[Function Router]}
  store={[Function Store]}
>
  <section>
    <mat-card>
      <mat-card-title>
        <mat-toolbar
          color="primary"
        >
          SIGN-IN
        </mat-toolbar>
      </mat-card-title>
      <mat-card-content>
        <form
          class="ng-untouched ng-pristine ng-invalid"
          novalidate=""
        >
          <div>
            <mat-form-field>
              <input
                class="ng-untouched ng-pristine ng-invalid"
                matinput=""
                name="login"
                ng-reflect-model=""
                ng-reflect-name="login"
                ng-reflect-required=""
                ngmodel=""
                placeholder="LOGIN"
                required=""
                type="text"
              />
              <mat-icon
                matsuffix=""
              >
                account_circle
              </mat-icon>
            </mat-form-field>
          </div>
          <div>
            <mat-form-field>
              <input
                class="ng-untouched ng-pristine ng-invalid"
                matinput=""
                name="password"
                ng-reflect-model=""
                ng-reflect-name="password"
                ng-reflect-required=""
                ngmodel=""
                placeholder="PASSWORD"
                required=""
                type="password"
              />
              <mat-icon
                matsuffix=""
              >
                lock
              </mat-icon>
            </mat-form-field>
          </div>
          
          <p
            class="err-message"
          >
            ERROR-TEST
          </p>
          <div
            class="actions"
          >
            <button
              color="primary"
              mat-flat-button=""
              type="submit"
            >
               SIGN-IN 
            </button>
            <a
              href="/forgot-password"
              mat-button=""
              ng-reflect-router-link="/forgot-password"
            >
              FORGOT-PASSWORD
            </a>
          </div>
          <mat-divider />
          <button
            class="sign-up"
            mat-button=""
            type="button"
          >
             SIGN-UP 
          </button>
        </form>
      </mat-card-content>
    </mat-card>
  </section>
</app-login-form>
`;
